packages/vector_graphics_compiler/lib/src/svg/node.dart

```diff
@@ -341,3 +341,66 @@ class DeferredNode extends AttributedNode {
     concreteRef.build(builder, transform);
   }
 }
+
+/// A leaf node in the tree that represents inline text.
+///
+/// Leaf nodes get added with all paint and transform accumulations from their
+/// parents applied.
+class TextNode extends AttributedNode {
+  /// Create a new [TextNode] with the given [text].
+  TextNode(this.text, this.dx, this.dy, SvgAttributes attributes)
+      : super(attributes);
+
+  /// The text this node contains.
+  final String text;
+
+  /// The x coordinate of the starting point of the text baseline.
+  final double dx;
+
+  /// The y coordinate of the starting point of the text baseline.
+  final double dy;
+
+  Paint? _paint(Rect bounds, AffineMatrix transform) {
+    final Fill? fill = attributes.fill?.toFill(bounds, transform);
+    final Stroke? stroke = attributes.stroke?.toStroke(bounds, transform);
+    if (fill == null && stroke == null) {
+      return null;
+    }
+    return Paint(
+      blendMode: attributes.blendMode,
+      fill: fill,
+      stroke: stroke,
+    );
+  }
+
+  TextConfig _textConfig(AffineMatrix transform) {
+    transform = transform.multiplied(attributes.transform);
+    return TextConfig(
+        text,
+        Point(dx, dy),
+        attributes.fontFamily ?? '',
+        int.tryParse(attributes.fontWeight ?? '') ?? 500,
+        double.tryParse(attributes.fontSize ?? '') ?? 16,
+        transform);
+  }
+
+  @override
+  AttributedNode applyAttributes(SvgAttributes newAttributes) {
+    return TextNode(
+      text,
+      dx,
+      dy,
+      attributes.applyParent(newAttributes),
+    );
+  }
+
+  @override
+  void build(DrawCommandBuilder builder, AffineMatrix transform) {
+    // TODO - bounds should come from the first parent viewbox.
```

Comment:
That might not work right? 

Imagine you have a parent box whose bounds include this text and a shape. You want to apply a gradient _only to_ the text. Your bounds will stretch the gradient wrong if it's object boundign box units.

Reply:
my reading of x,y % units was that the size came from the nearest object with a shape - text doesn't have a size before being laid out right?

Reply:
Hm. I think you're right, or I'm just getting this wrong in flutter_svg anyway. We can just go with this and try to fix it later if it's wrong - I suspect that whatever might be wrong here is wrong in flutter_svg too though.

Reply:
We need to fix this for all shapes - relative sizing does not work right now